lib/compress.js

```diff
@@ -2582,6 +2598,18 @@ merge(Compressor.prototype, {
 
     OPT(AST_Call, function(self, compressor){
         var exp = self.expression;
+        if (compressor.option("reduce_vars")
+            && exp instanceof AST_SymbolRef) {
+            var def = exp.definition();
+            if (def.fixed) {
+                exp = def.fixed;
+                if (compressor.option("unused")
+                    && def.references.length == 1
+                    && exp instanceof AST_Function) {
```

Comment:
So what I have done is to populate `SymbolDef.fixed` with [a clone of `AST_Defun` but as `AST_Function`](https://github.com/mishoo/UglifyJS2/pull/1560/files#diff-230c53e13e3ee367aec7d49470eb42caR269), the field for which `reduce_vars` use to track the singly assigned value that is safe to be used.